Proposal of features such as Net classes, Date and a new theme.

Rollbacke opened the pull request:
Add RAM free, Disk used percent, Class Net, Class Date and bash-dark-green theme.

Rollbacke commented:
All themes are filled. Normally, I tested everything.

mathoudebine commented:
Hello @Rollbacke 
Thanks for this PR!
I see that you and @RussNelson opened almost the same PR, so i will not be able to merge both.

@RussNelson can you confirm the PR #65 and #67 are from @Rollbacke's code?
If so, would it be OK to close them and merge only this PR which has both?

mathoudebine commented on library/stats.py:
```diff
@@ -532,3 +571,195 @@ def stats():
+class Net:
+    @staticmethod
+    def stats():
+        pnic_before_file = "/tmp/pnic_before"
```
Is it possible to avoid using an external file?

Rollbacke replied:
Sorry, I haven't found how to do it yet.

Rollbacke replied:
Found how to do without temporary file.
I pushed the commit (below).

Rollbacke commented:
> Hello @Rollbacke Thanks for this PR! I see that you and @RussNelson opened almost the same PR, so i will not be able to merge both.
> 
> @RussNelson can you confirm the PR #65 and #67 are from @Rollbacke's code? If so, would it be OK to close them and merge only this PR which has both?

Hi @mathoudebine,
These are indeed my proposals that I had put in the Issues.
I cleaned up and improved the code.
From what I suggested:
* for the Net class, contained unnecessary lines and lacked checks that can crash the script (if the interface doesn't exist, it doesn't work).
* for the Date class, since I wasn't using the strftime function, the date displayed "3-11-2022", which doesn't look pretty. With strftime, this is "03-11-2022".
In addition, I modified all of the themes (theme.yaml) to make it work.
I added the theme bash-dark-green I made myself.
I also added two metrics: RAM free, Disk used percent.

mathoudebine commented on res/themes/3.5inchTheme2/theme.yaml:
```diff
@@ -279,6 +288,15 @@ STATS:
         BACKGROUND_COLOR: 132, 154, 165
         # BACKGROUND_IMAGE: background.png
+      PERCENT_TEXT:
+        SHOW: True
```
SHould be false for this theme, otherwise there is a text over the disk icon

mathoudebine commented on res/themes/bash-dark-green/theme.yaml:
```diff
@@ -0,0 +1,401 @@
+        BACKGROUND_IMAGE: background.png
+  NET:
+    INTERVAL: 1
+    WLO1:
```
WLO1 -> WL0

mathoudebine commented on res/themes/bash-dark-green/theme.yaml:
```diff
@@ -0,0 +1,401 @@
+          FONT_COLOR: 53, 191, 92
+          # BACKGROUND_COLOR: 0, 0, 0
+          BACKGROUND_IMAGE: background.png
+    ECRANDOCK:
```
-> ETH

mathoudebine commented on res/themes/bash-dark-green/theme.yaml:
```diff
@@ -0,0 +1,401 @@
+---
+display:
+  # Specify the display orientation for this theme: portrait, landscape, reverse_portrait, reverse_landscape
+  DISPLAY_ORIENTATION: reverse_portrait
```
can we commit this as portrait by default? This way it would fit most people orientation 👍

Rollbacke commented:
Oh. Sorry about that. I will fix that.

Rollbacke commented:
It's OK ;)

mathoudebine commented:
Thanks @Rollbacke for this awesome work! PR has been merged and a new version will be created with your work 🚀
